src/components/PostDetails.tsx

```diff
@@ -4,69 +4,51 @@
 
 import * as commentsApi from '../api/comments';
 
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { CommentData } from '../types/Comment';
+import { useSelector } from 'react-redux';
+import { setError } from '../features/post/postsSlice';
+import {
+  loadCommentsAsync,
+  selectComments,
+  selectHasError,
+  selectLoadedComment,
+  setComments,
+  setLoadedComment,
+} from '../features/comments/commentsSlice';
+import { useAppDispatch } from '../app/hooks';
+import { selectSelectedPost } from '../features/selectedPost/selectedPostSlice';
+
+export const PostDetails: React.FC = () => {
+  const dispatch = useAppDispatch();
+
+  const post = useSelector(selectSelectedPost);
+  const comments = useSelector(selectComments);
+  const loaded = useSelector(selectLoadedComment);
+  const hasError = useSelector(selectHasError);
   const [visible, setVisible] = useState(false);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
+    dispatch(setLoadedComment(false));
+    dispatch(setError(false));
     setVisible(false);
-    setError(false);
 
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+    if (post) {
+      dispatch(loadCommentsAsync(post.id));
     }
-  };
-
-  useEffect(() => {
-    loadComments();
-  }, []);
+  }
 
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+  useEffect(loadComments, [post?.id]);
 
   const addComment = async ({ name, email, body }: CommentData) => {
     try {
       const newComment = await commentsApi.createComment({
         name,
         email,
         body,
-        postId: post.id,
+        postId: post ? post.id : 0,
       });
 
-      setComments(currentComments => [...currentComments, newComment]);
+      setComments([...comments, newComment]);
```

Comment:
The `setComments` function should be dispatched as an action to update the Redux state. Replace `setComments([...comments, newComment]);` with `dispatch(setComments([...comments, newComment]));` to ensure that the state is updated correctly in the Redux store.